### ISSUE TITLE
Not identified as monospaced font

### ISSUE THREAD

**Issue by iget-master** [madmalik/mononoki#23]
Tried to use this font on PHPStorm over Ubuntu 14.04, installed the latest .ttf fonts, and it's not listed as a monospaced font.

**Comment by madmalik** [comment]
I'll look into that. I'm sure i have a ubuntu virtualbox image somewhere ^^

**Comment by iget-master** [comment]
Thanks @madmalik :)

**Comment by kharaktur** [comment]
I have a similar issue, using PyCharm Community Ed 2016.1.3 this beautiful font isn't listed as monospaced

**Comment by glebd** [comment]
It could also be JRE issue as both PHPStorm and PyCharm are Java-based (IntelliJ IDEA-based). On the Mac CLion (another IDEA-based IDE from JetBrains) correctly sees Mononoki as a monospaced font.

**Comment by madmalik** [comment]
Time to take advantage of my student status and get some IntelliJ IDEs for testing

**Comment by kharaktur** [comment]
Thanks for your quick reply. No need to spend a dime to test this if you like, pycharm community is free. I am using it on windows. BTW I have tried several fonts in the ide, Deja Vu sans mono was my favorite but Mononoki is both beautiful to look at and super easy (for my old eyes) to read. Super job.
 

    On Saturday, May 21, 2016 3:04 PM, madmalik <notifications@github.com> wrote:
 

 Time to take advantage of my student status and get some IntelliJ IDEs for testing—

**Comment by meilke** [comment]
Same on Linux for Sublime text 3. I used this guide to install the font: https://community.linuxmint.com/tutorial/view/29.

**Comment by AnwarShah** [comment]
It's working on Sublime text 3

**Comment by meilke** [comment]
@AnwarShah you are right, I used `"font_face": "mononoki-Regular"` instead of `"font_face": "mononoki". Now it is working.

**Comment by DmitriK** [comment]
Does not show up as a monospaced font in KDE system-wide fixed width or Konsole font selection.

**Comment by darkwise** [comment]
@madmalik I wonder if mononoki font miss a monospace subfamily like having italic or bold. The fact that  this font is though monoscpaced doesn't mean it's a monospace one. I think software look for **openTypeNamePreferredSubfamilyName** having **monospace**.

So what do you think about having a "symlinking" regular sources with a  [new monospace subfamily] (https://github.com/madmalik/mononoki/blob/master/src/mononoki-Regular.ufo/fontinfo.plist) ? 

Cheers,

**Comment by madmalik** [comment]
Mhh, I think you are right. 

I'll test that and make a release if all plattforms detect mononoki as monospace after that.


Thank you!

**Comment by sib-null** [comment]
I'm affected by this problem as well, on Manjaro KDE.  Let me know if you need any help verifying the fix.

**Comment by sarbash** [comment]
It has reached Debian/sid and has the same issue with Konsole. Fontmatrix, Libreoffice, GIMP see it.
Konsole doesn't.

**Comment by jomendoz** [comment]
@DmitriK @sib-null @sarbash Have you found a way to use the font with Konsole? I tried sym-linking files to /usr/share/fonts/TTF/ but didn't work.

**Comment by DmitriK** [comment]
@jomendoz You'll want to edit ~/.local/share/konsole/PROFILE_NAME.profile and change the first entry in the Font list to mononoki.

**Comment by jomendoz** [comment]
@DmitriK Thank you very much! This worked perfectly.

**Comment by sarbash** [comment]
@DmitriK I've found a GUI-way.

In Profile Settings, tick "Show All Fonts" checkbox - viola! I just am wondering why it isn't recognized as a monospaced font in Konsole.

**Comment by sarbash** [comment]
Also Firefox doesn't see it as a monospaced font.

**Comment by jeberger** [comment]
I've had a similar issue with some customizations I've made to the Hack font. I've found that in order to be recognized as monospace everywhere, a font needs two things:

- Every character must be the same width (and I do mean _every_ character).
- The font must have the proper OS/2 Panose setting for "proportions" (with the default configuration, FontForge will set it automagically if all characters have the same width).

See https://fontforge.github.io/faq.html#monospace

**Comment by exvx** [comment]
Related: https://github.com/madmalik/mononoki/issues/59

**Comment by csauve** [comment]
I'm trying to use Mononoki as my terminal font in kitty, but it is always being ignored because kitty seems to validate that candidate fonts are indeed monospaced. When I use `fc-list :spacing=100`, Mononoki is not listed as a result whereas other monospaced fonts installed on my system are.

I am wondering if this is caused by [this line of fontinfo.plist](https://github.com/madmalik/mononoki/blob/master/src/mononoki-Regular.ufo/fontinfo.plist#L13). According to [this page](http://unifiedfontobject.org/versions/ufo3/fontinfo.plist/), this key should be `postscriptIsFixedPitch` and not `isFixedPitched`.

I couldn't verify making this change works because I couldn't successfully build the fonts from this repo using Google's fontmake (something about "locl" feature not being defined). Maybe someone more experienced in font building or @madmalik can give it a try?

**Comment by gambhiro** [comment]
Thank for the cool terminal font! I like the double-story g, the narrower width but tall x-height which helps reading CamelCase names.

Same issue as @csauve . Kitty terminal won't use Mononoki, and defaults to the system default monospace font.

It does work however when setting the system default monospace to Mononoki.

In `kitty.conf`:

```
font_family Mononoki
font_size 10.0
```

I downloaded the Mononoki release archive and extracted to `~/.fonts`, then updated the cache with

```
$ fc-cache -f ~/.fonts
```

The system does find the font by name, but it is not recognized as a monospace font:

```
$ fc-match "Mononoki"
mononoki-Regular.ttf: "mononoki" "Regular"

$ fc-list :spacing=100 | grep -i mononoki
(empty)
```

Set the Gnome setting:

```
gsettings set org.gnome.desktop.interface monospace-font-name "Mononoki 11"
```

This made it work in apps such as `gedit`.

For terminals and so on, also edit `~/.config/fontconfig/fonts.conf`:

```
<match target="pattern">
	<test name="family" qual="any">
		<string>monospace</string>
	</test>
	<edit binding="strong" mode="prepend" name="family">
		<string>Mononoki</string>
	</edit>
</match>
```

(from [stackoverflow](https://unix.stackexchange.com/a/106170/2528))

Then I ran `fc-cache -f ~/.fonts` again.

It worked when opening a new kitty terminal. It did not need re-login.

**Comment by csauve** [comment]
I was successfully able to make this font detected as monospaced by doing the following:

1. made the edit in [my previous post](#issuecomment-445572011)
2. opened the .ufo directory in [fontforge](https://fontforge.github.io/)
3. I'm not sure if this step mattered, but selecting all glyphs, right clicked, and set all widths to 575. I didn't trust [the validation](https://fontforge.github.io/faq.html#monospace) because even when I set one glyph to 574, no problems were detected by fontforge
4. Exported the font to TTF into ~/.local/share/fonts/mononoki/
5. Ran `fc-cache -f` to rebuild the font cache

The font now shows up as monospace when I run `fc-list :spacing=100` and works in kitty.
